private/pkg/protostat/protostatos/file_walker.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package protostatos
```

Comment:
Out of curiosity, why break these out into two sub-packages, vs. having two functions in protostat to provide the two implementations?

Reply:
Honestly it's a little overkill - I think it's academically a bit cleaner (if you aren't using the storage package, you can just depend on os, and in theory vice versa), but it may be a little much here. In a lot of ways, it's just copying the style that already exists in the codebase, which I think takes precedence - consistency comes first. We can revisit this though, I think I got a little out of hand here.